docs/src/man/joins_new.md

```diff
@@ -0,0 +1,879 @@
+  994 │   993     30  male        3  own      little           little          ⋯
+  995 │   994     50  male        2  own      missing          missing
+  996 │   995     31  female      1  own      little           missing
+  997 │   996     40  male        3  own      little           little
+  998 │   997     38  male        2  own      little           missing         ⋯
+  999 │   998     23  male        2  free     little           little
+ 1000 │   999     27  male        2  own      moderate         moderate
+                                                  4 columns and 985 rows omitted
+
+julia> german = view(german_ref, 1:10, 1:5)
+10×5 SubDataFrame
+ Row │ id     Age    Sex     Job    Housing
+     │ Int64  Int64  String  Int64  String
+─────┼──────────────────────────────────────
+   1 │     0     67  male        2  own
+   2 │     1     22  female      2  own
+   3 │     2     49  male        1  own
+   4 │     3     45  male        2  free
+   5 │     4     53  male        2  free
+   6 │     5     35  male        1  free
+   7 │     6     53  male        2  own
+   8 │     7     35  male        3  rent
+   9 │     8     61  male        1  own
+  10 │     9     28  male        3  own
+```
+
+This table has all the `:id`s of all factors to make comparisions for risk factor. Let’s say we 
+have another data frame, which has the risk (good or bad) of these individuals:
+
+```jldoctest dataframe
+julia> risk_ref = CSV.read(joinpath(dirname(pathof(DataFrames)),
+                                    "..", "docs", "src", "assets", "risk.csv"),
+                           DataFrame);
+
+julia> risk = view(risk_ref, 1:10, :)
+10×2 SubDataFrame
+ Row │ id     Risk
+     │ Int64  String
+─────┼───────────────
+   1 │     0  good
+   2 │     1  bad
+   3 │     2  good
+   4 │     3  good
+   5 │     4  bad
+   6 │     5  good
+   7 │     6  good
+   8 │     7  good
+   9 │     8  good
+  10 │     9  bad
+```
+
+We now have 2 data frames:
+- `german_ref` holds the `:id`s, `:Age`, `:Sex`, `:Job` and `:Housing`
+- `risk_ref` holds the `:id`s and `:Risk`.
+
+In relational database theory, to combine two datasets referred to as a join. The columns used to 
+determine which rows should be combined during a join are called keys.
+
+The following functions are provided to perform seven kinds of joins:
+
+- `innerjoin`: the output contains rows for values of the key that exist in all passed data frames.
+- `leftjoin`: the output contains rows for values of the key that exist in the first (left) argument, 
+  whether or not that value exists in the second (right) argument.
+- `rightjoin`: the output contains rows for values of the key that exist in the second (right) argument, 
+  whether or not that value exists in the first (left) argument.
+- `outerjoin`: the output contains rows for values of the key that exist in any of the passed data frames.
+- `semijoin`: Like an inner join, but output is restricted to columns from the first (left) argument.
```

Comment:
where have you taken this description from? We use another definition based on subset or rows(which is not 100% equivalent)